src/pde_solve.jl

```diff
@@ -59,5 +66,80 @@ function DiffEqBase.solve(
 
     Flux.train!(loss, ps, data, opt; cb = cb)
 
-    save_everystep ? iters : u0(X0)[1]
+
+    if give_limit == false
+        save_everystep ? iters : u0(X0)[1]
+    else
+        A = prob.A
+        u_domain = prob.u_domain
+
+        ## UPPER LIMIT
+        sdeProb = SDEProblem(μ , σ , X0 , prob.tspan)
+        ensembleprob = EnsembleProblem(sdeProb)
+        sim = solve(ensembleprob, sdealg, ensemblealg, dt=dt,trajectories=trajectories_upper,adaptive=false,prob.kwargs...)
```

Comment:
```suggestion
        sim = solve(ensembleprob, EM(), ensemblealg, dt=dt, trajectories=trajectories_upper,prob.kwargs...)
```

I think for NNPDEHan it makes sense to hardcode it as `EM` since the method itself is essentially all about Euler-Maruyama discretizations. For the other deep BSDE method we should do it more generically, but that can follow in a different PR.